util/merkle/src/triedb.rs

```diff
@@ -40,12 +40,12 @@ use crate::{Trie, TrieError};
 ///
 /// let mut memdb = MemoryDB::new();
 /// let mut root = H256::new();
-/// TrieDBMut::new(&mut memdb, &mut root).insert(b"foo", b"bar").unwrap();
-/// let t = TrieDB::try_new(&memdb, &root).unwrap();
+/// TrieFactory::create(&mut memdb, &mut root).insert(b"foo", b"bar").unwrap();
+/// let t = TrieFactory::readonly(&memdb, &root).unwrap();
 /// assert!(t.contains(b"foo").unwrap());
 /// assert_eq!(t.get(b"foo").unwrap().unwrap(), b"bar".to_vec());
 /// ```
-pub struct TrieDB<'db> {
+pub(crate) struct TrieDB<'db> {
```

Comment:
Is this required?

Reply:
Yes. The TrieDB should not be exported out of the crate, but it's used inside the crate.

Reply:
Isn't it exported only when we `pub use` in the `mod.rs`?
I think removing `pub` in `mod.rs` is enough.

Reply:
I want to guarantee that this doesn't go out of the crate.

Reply:
Hmm. I think that we didn't "have to" introduce this change.
You can leave it as is if you want, but I'd like to suggest minimizing the change if possible.